File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLPostAddress.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.TLContext;
import com.github.badoualy.telegram.tl.core.TLObject;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;

import static com.github.badoualy.telegram.tl.StreamUtils.readTLString;
import static com.github.badoualy.telegram.tl.StreamUtils.writeString;
import static com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID;
import static com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize;

/**
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public class TLPostAddress extends TLObject {

    public static final int CONSTRUCTOR_ID = 0x1e8caaeb;

    protected String streetLine1;

    protected String streetLine2;

    protected String city;

    protected String state;

    protected String countryIso2;

    protected String postCode;

    private final String _constructor = "postAddress#1e8caaeb";

    public TLPostAddress() {
    }

    public TLPostAddress(String streetLine1, String streetLine2, String city, String state, String countryIso2, String postCode) {
        this.streetLine1 = streetLine1;
        this.streetLine2 = streetLine2;
        this.city = city;
        this.state = state;
        this.countryIso2 = countryIso2;
        this.postCode = postCode;
    }

    @Override
    public void serializeBody(OutputStream stream) throws IOException {
        writeString(streetLine1, stream);
        writeString(streetLine2, stream);
        writeString(city, stream);
        writeString(state, stream);
        writeString(countryIso2, stream);
        writeString(postCode, stream);
    }

    @Override
    @SuppressWarnings({"unchecked", "SimplifiableConditionalExpression"})
    public void deserializeBody(InputStream stream, TLContext context) throws IOException {
        streetLine1 = readTLString(stream);
        streetLine2 = readTLString(stream);
        city = readTLString(stream);
        state = readTLString(stream);
        countryIso2 = readTLString(stream);
        postCode = readTLString(stream);
    }

    @Override
    public int computeSerializedSize() {
        int size = SIZE_CONSTRUCTOR_ID;
        size += computeTLStringSerializedSize(streetLine1);
        size += computeTLStringSerializedSize(streetLine2);
        size += computeTLStringSerializedSize(city);
        size += computeTLStringSerializedSize(state);
        size += computeTLStringSerializedSize(countryIso2);
        size += computeTLStringSerializedSize(postCode);
        return size;
    }

    @Override
    public String toString() {
        return _constructor;
    }

    @Override
    public int getConstructorId() {
        return CONSTRUCTOR_ID;
    }

    public String getStreetLine1() {
        return streetLine1;
    }

    public void setStreetLine1(String streetLine1) {
        this.streetLine1 = streetLine1;
    }

    public String getStreetLine2() {
        return streetLine2;
    }

    public void setStreetLine2(String streetLine2) {
        this.streetLine2 = streetLine2;
    }

    public String getCity() {
        return city;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public String getState() {
        return state;
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getCountryIso2() {
        return countryIso2;
    }

    public void setCountryIso2(String countryIso2) {
        this.countryIso2 = countryIso2;
    }

    public String getPostCode() {
        return postCode;
    }

    public void setPostCode(String postCode) {
        this.postCode = postCode;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLPostAddress.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.core.TLObject
import com.github.badoualy.telegram.tl.serialization.TLDeserializer
import com.github.badoualy.telegram.tl.serialization.TLSerializer
import java.io.IOException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.jvm.Throws

/**
 * postAddress#1e8caaeb
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
class TLPostAddress() : TLObject() {
    var streetLine1: String = ""

    var streetLine2: String = ""

    var city: String = ""

    var state: String = ""

    var countryIso2: String = ""

    var postCode: String = ""

    private val _constructor: String = "postAddress#1e8caaeb"

    override val constructorId: Int = CONSTRUCTOR_ID

    constructor(
            streetLine1: String,
            streetLine2: String,
            city: String,
            state: String,
            countryIso2: String,
            postCode: String
    ) : this() {
        this.streetLine1 = streetLine1
        this.streetLine2 = streetLine2
        this.city = city
        this.state = state
        this.countryIso2 = countryIso2
        this.postCode = postCode
    }

    @Throws(IOException::class)
    override fun serializeBody(tlSerializer: TLSerializer) = with (tlSerializer)  {
        writeString(streetLine1)
        writeString(streetLine2)
        writeString(city)
        writeString(state)
        writeString(countryIso2)
        writeString(postCode)
    }

    @Throws(IOException::class)
    override fun deserializeBody(tlDeserializer: TLDeserializer) = with (tlDeserializer)  {
        streetLine1 = readString()
        streetLine2 = readString()
        city = readString()
        state = readString()
        countryIso2 = readString()
        postCode = readString()
    }

    override fun computeSerializedSize(): Int {
        var size = SIZE_CONSTRUCTOR_ID
        size += computeTLStringSerializedSize(streetLine1)
        size += computeTLStringSerializedSize(streetLine2)
        size += computeTLStringSerializedSize(city)
        size += computeTLStringSerializedSize(state)
        size += computeTLStringSerializedSize(countryIso2)
        size += computeTLStringSerializedSize(postCode)
        return size
    }

    override fun toString() = _constructor

    override fun equals(other: Any?): Boolean {
        if (other !is TLPostAddress) return false
        if (other === this) return true

        return streetLine1 == other.streetLine1
                && streetLine2 == other.streetLine2
                && city == other.city
                && state == other.state
                && countryIso2 == other.countryIso2
                && postCode == other.postCode
    }
    companion object  {
        const val CONSTRUCTOR_ID: Int = 0x1e8caaeb.toInt()
    }
}
